### PR TITLE
Add arianna

### PR DESCRIPTION
**Pull request by CarlSchwan**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

Not that this currently build from the latest master, I will do the initial release tomorrow or on Tuesday but I wanted to start early with the flatpak submission process

### CODE REVIEW

**Comment by hfiguiere** [comment]
cc @flathub/kde

File: org.kde.arianna.json [hunk 1]
```diff
@@ -0,0 +1,94 @@
+{
+    "id": "org.kde.arianna",
+    "branch": "master",
```
Comment (hfiguiere) [inline org.kde.arianna.json:3]:
```suggestion
```

File: org.kde.arianna.json [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+    "runtime-version": "5.15-22.08",
+    "sdk": "org.kde.Sdk",
+    "base": "io.qt.qtwebengine.BaseApp",
+    "base-version": "5.15-22.08",
+    "command": "arianna",
+    "finish-args": [
+        "--share=network",
+        "--share=ipc",
+        "--socket=x11",
+        "--socket=wayland",
```
Comment (hfiguiere) [inline org.kde.arianna.json:14]: if is supports wayland, then it's `fallback-x11`

File: org.kde.arianna.json [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+        "--device=dri",
+        "--filesystem=host:ro",
```
Comment (hfiguiere) [inline org.kde.arianna.json:16]: this is a broad filesystem permission

File: org.kde.arianna.json [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+        "--env=BALOO_DB_PATH=.local/share/baloo",
```
Comment (hfiguiere) [inline org.kde.arianna.json:17]: it doesn't use the standard directories?

File: org.kde.arianna.json [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+                "prefix=/app"
+            ],
+            "subdir": "libraries/liblmdb"
+        },
+        {
+            "config-opts": [
+                "-DBUILD_TESTING=OFF",
+                "-DCMAKE_INSTALL_LIBDIR=lib"
+            ],
+            "name": "baloo",
```
Comment (hfiguiere) [inline org.kde.arianna.json:65]: usually `name` is first.

File: org.kde.arianna.json (hunk 3 above)
Comment (CarlSchwan) [reply org.kde.arianna.json:16]: Yes, it's needed Arianna will search and index epubs in the home folder (using Baloo)

File: org.kde.arianna.json [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+                        "project-id": 242933,
+                        "stable-only": true,
+                        "url-template": "https://download.kde.org/stable/kirigami-addons/kirigami-addons-$version.tar.xz"
+                    }
+                }
+            ]
+        },
+        {
+            "name": "lmdb",
+            "sources": [
```
Comment (hfiguiere) [inline org.kde.arianna.json:41]: usually `sources` is last.

**Comment by hfiguiere** [comment]
bot, build org.kde.arianna

File: org.kde.arianna.json (hunk 3 above)
Comment (hfiguiere) [reply org.kde.arianna.json:16]: home is not host.

File: org.kde.arianna.json (hunk 4 above)
Comment (CarlSchwan) [reply org.kde.arianna.json:17]: It's configured like elisa https://github.com/flathub/org.kde.elisa/blob/master/org.kde.elisa.json#L20

File: org.kde.arianna.json (hunk 6 above)
Comment (CarlSchwan) [reply org.kde.arianna.json:45]: Done

File: org.kde.arianna.json (hunk 3 above)
Comment (CarlSchwan) [reply org.kde.arianna.json]: Ok changed

Comment (hfiguiere) [reply org.kde.arianna.json]: does that mean that if baloo isn't installed on the host this won't work?

**Comment by hfiguiere** [comment]
bot, build org.kde.arianna

**Comment by CarlSchwan** [comment]
No idea why it fails, it seems Anitya is failing to check that lmdb 0.9.30 is not existing

File: org.kde.arianna.json (hunk 3 above)
Comment (CarlSchwan) [reply org.kde.arianna.json]: There is a fallback for other DE, android and windows whic doesn't use baloo.

File: org.kde.arianna.json [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+            "name": "lmdb",
+            "no-autogen": true,
+            "make-install-args": [
+                "prefix=/app"
+            ],
+            "subdir": "libraries/liblmdb",
+            "sources": [
+                {
+                    "type": "archive",
+                    "url": "https://github.com/LMDB/lmdb/archive/refs/tags/LMDB_0.9.29.tar.gz",
```
Comment (tsdgeos) [inline org.kde.arianna.json:47]: You probably want ldap url

org.kde.elisa/org.kde.elisa.json:                    "url": "https://git.openldap.org/openldap/openldap/-/archive/LMDB_0.9.30/openldap-LMDB_0.9.30.tar.gz",
org.kde.elisa/org.kde.elisa.json:                        "url-template": "https://git.openldap.org/openldap/openldap/-/archive/LMDB_$version/openldap-LMDB_$version.tar.gz"

Comment (CarlSchwan) [reply org.kde.arianna.json]: Thanks

File: org.kde.arianna.json [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+                }
+            ]
+        },
+        {
+            "name": "arianna",
+            "buildsystem": "cmake-ninja",
+            "sources": [
+                {
+                    "type": "git",
+                    "url": "https://invent.kde.org/graphics/arianna.git",
```
Comment (tsdgeos) [inline org.kde.arianna.json:86]: No release tarball? This is really not something we do in other places.

Comment (CarlSchwan) [reply org.kde.arianna.json:86]: See the description of this MR, I'm gonna release the initial release today. But I wanted to have a flatpak release around the same time as the release.

**Comment by travier** [comment]
LGTM. As it's a new app, it's OK if we don't have a tarball yet.

Need investigation for the lmdb check. I've seen that somewhere else already

**Comment by CarlSchwan** [comment]
bot, build org.kde.arianna

**Comment by CarlSchwan** [comment]
bot, build org.kde.arianna

**Comment by CarlSchwan** [comment]
bot, build org.kde.arianna

**Comment by travier** [comment]
Does the check pass locally? We have a similar check in another KDE Flatpak I think.

**Comment by CarlSchwan** [comment]
> Does the check pass locally? We have a similar check in another KDE Flatpak I think.

the hash is broken and I have a fix for that locally. Unfortunately, baloo 5.105 requires the latest framework release. I saw you published https://invent.kde.org/packaging/flatpak-kde-runtime/-/merge_requests/121 so I'm now waiting for it to be merged ;)

**Comment by CarlSchwan** [comment]
bot, build org.kde.arianna

**Comment by CarlSchwan** [comment]
bot, build org.kde.arianna

**Comment by CarlSchwan** [comment]
bot, build org.kde.arianna

**Pull request closed without merging**